### ISSUE TITLE
Killing torxakis doesn't kill the server or the constraint server

### ISSUE THREAD

**Issue by pjljvandelaar** [TorXakis/TorXakis#365]
Added by Petra van den Bos 

When killing (ctr+c) torxakis, i.e. the user interface, the processes of the server and the constraint solver stay alive.

Scenario: I kill (ctr+c) the command prompt in which TorXakis runs (because the constraint solver is taking too long to evaluate string expressions). The Task Manager then shows that the CVC4-process is still running.
Sugested solution: When TorXakis receives a kill, TorXakis first kills the constraint solver, and only after that it kills itself.

**Comment by pjljvandelaar** [comment]
See 
https://hackage.haskell.org/package/process-1.4.3.0/docs/System-Process.html
terminateProcess and interruptProcessGroupOf
On Windows systems, it generates a CTRL_BREAK_EVENT and will only work for processes created using createProcess and setting the 
[create_group](https://hackage.haskell.org/package/process-1.4.3.0/docs/System-Process.html#v:create_group) flag

**Comment by capitanbatata** [comment]
Now, when the `txsserver` is started from the `txsui` process, Ctrl-C kills the txsserver and the solver processes. So this issue boils down to properly handling interruptions in the server, in case the server is started as a separate process. @pjljvandelaar should we close this after merging `feat/start-txsserver-from-ui`?

**Comment by pjljvandelaar** [comment]
What about closing and making a low priority issue for the server as a separate process

**Comment by capitanbatata** [comment]
Next week I plan to work on the issue of proper exception handling at the `txsserver`, namely #55. So I guess no need to make a separate issue.
